### PR TITLE
wrap runPython in async

### PR DESCRIPTION
**Pull request by madhur-tandon**
CC: @JeffersGlass @antocuni @hoodmane 

TODO comments still required at two places.

Thing to consider: In `pyrepl.ts` -- should the call to `pyExec` be inside a mutex lock? Just the way it was done for the `pyExec` call in `pyscript.ts`

### CODE REVIEW

File: pyscriptjs/src/pyodide.ts [hunk 1]
```diff
@@ -78,12 +78,15 @@ export class PyodideInterpreter extends Interpreter {
...
-        this.run('print("Python initialization complete")')
+        await this.run('print("Python initialization complete")')
     }
 
-    run(code: string): unknown {
-        return this.interface.runPython(code);
+    /* eslint-disable */
+    async run(code: string): Promise<{result: any}> {
+        // TODO: ADD A COMMENT TO EXPLAIN WHY RESULT IS WRAPPED IN A DICTIONARY
+        return { result: this.interface.runPython(code) };
```
Comment (antocuni) [inline pyscriptjs/src/pyodide.ts:87]: yes please, add a comment :)
Writing it here so that we don't forget :)

File: pyscriptjs/src/main.ts [hunk 2]
```diff
@@ -65,6 +64,7 @@ export class PyScriptApp {
     PyScript: ReturnType<typeof make_PyScript>;
     plugins: PluginManager;
     _stdioMultiplexer: StdioMultiplexer;
+    tagExecutionLock: any;
```
Comment (antocuni) [inline pyscriptjs/src/main.ts:67]:
```suggestion
    tagExecutionLock: any; // this is used to ensure that py-script tags are executed sequentally
```

**Review by antocuni — COMMENTED** [review]
LGTM, thanks!

> should the call to pyExec be inside a mutex lock? Just the way it was done for the pyExec call in pyscript.ts

I *think* it is not needed

File: pyscriptjs/src/main.ts (hunk 2 above)
Comment (madhur-tandon) [reply pyscriptjs/src/main.ts]: done!

File: pyscriptjs/tests/integration/test_01_basic.py [hunk 3]
```diff
@@ -311,6 +311,7 @@ def test_getPySrc_returns_source_code(self):
             == 'print("hello world!")\n'
         )
 
+    @pytest.mark.skip(reason="pys-onClick is broken, we should kill it")
```
[the remark this answers is not in the document]
Comment (madhur-tandon) [reply pyscriptjs/tests/integration/test_01_basic.py]: done!

[the remark this answers is not in the document]
Comment (madhur-tandon) [reply pyscriptjs/tests/integration/test_01_basic.py]: See https://github.com/pyscript/pyscript/issues/1213

File: pyscriptjs/src/pyodide.ts (hunk 1 above)
Comment (madhur-tandon) [reply pyscriptjs/src/pyodide.ts:104]: done!

File: pyscriptjs/src/main.ts [hunk 4]
```diff
@@ -225,12 +226,12 @@ export class PyScriptApp {
...
 
         // import some carefully selected names into the global namespace
-        await interpreter.run(`
+        (await interpreter.run(`
         import js
         import pyscript
         from pyscript import Element, display, HTML
         pyscript._install_deprecated_globals_2022_12_1(globals())
-        `);
+        `)).result;
```
Comment (hoodmane) [inline pyscriptjs/src/main.ts:234]: This isn't used though?

Comment (madhur-tandon) [reply pyscriptjs/src/main.ts]: thanks, removed!

**Review by antocuni — APPROVED** [review]

**Pull request merged**